.github/workflows/deploy.yml

```diff
@@ -8,84 +8,202 @@ on:
         type: environment
         required: true
 jobs:
-  deploy:
+  perpare-deployment:
     environment: ${{ inputs.environment }}
-    name: Deploy Snap and UI
-    runs-on: ubuntu-20.04
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
```

Comment:
should we fix the version instead of using ubuntu-latest to control the image version we use ?

Reply:
may be fixing a version is better, lets move it to static version then

Reply:
i just find that it is using ubuntu-latest everywhere, and only this doc is using ubuntu-20.04 
so may be to keep it safe , use ubuntu-latest?

Reply:
ok